lib/types/plans.d.ts

```diff
@@ -0,0 +1,87 @@
+import { IMap, RazorpayPaginationOptions, INormalizeError } from "./api";
+import { Items } from "./items";
+
+export declare namespace Plans {
+    interface RazorpayPlanBaseRequestBody {
+        /**
+         * Details of the plan.
+         */
+        item: Items.RazorpayItemBaseRequestBody;
+        /**
+         * This, combined with interval, defines the frequency. Possible values:
+         * `daily`, `weekly`, `monthly`, `yearly`
+         * 
+         * If the billing cycle is 2 months, the value should be monthly.
+         */
+        period: "daily" | "weekly" | "monthly" | "yearly";
+        /**
+         * This, combined with `period`, defines the frequency. 
+         * If the billing cycle is 2 months, the value should be `2`.
+         */
+        interval: number;
+        /**
+         * Notes you can enter for the contact for future reference. 
+         * This is a key-value pair. You can enter a maximum of 15 key-value pairs. 
+         * For example, `note_key`: `Beam me up Scotty`
+         */
+        notes?: IMap<string | number>;
+    }
+
+    interface RazorpayPlanCreateRequestBody extends RazorpayPlanBaseRequestBody { }
+
+    interface RazorPayPlans extends RazorpayPlanBaseRequestBody {
+        /**
+         * The unique identifier linked to a plan
+         */
+        id: string;
+        /**
+         * Indicates the type of entity.
+         */
+        entity: string;
+        /**
+         * The Unix timestamp at which the plan was created.
+         */
+        created_at: number;
+        /**
+         * Details of the plan.
+         */
+        item: Items.RazorpayItem;
+    }
+}
+
+declare function plans(api: any): {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏ @ankitdas13 Payments has Plans types